dbms/src/Functions/FunctionsString.cpp

```diff
@@ -5956,6 +5956,256 @@ class FunctionFormatDecimal : public IFunction
     }
 };
 
+class FunctionTiDBUnHex : public IFunction
+{
+public:
+    static constexpr auto name = "tidbUnHex";
+    FunctionTiDBUnHex() = default;
+
+    static FunctionPtr create(const Context & /*context*/)
+    {
+        return std::make_shared<FunctionTiDBUnHex>();
+    }
+
+    std::string getName() const override { return name; }
+    size_t getNumberOfArguments() const override { return 1; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (!arguments[0]->isStringOrFixedString() && !arguments[0]->isNumber())
```

Comment:
```suggestion
        if (!arguments[0]->isString() && !arguments[0]->isNumber())
```
Since we don't use fixedString.